instrumentation/elasticsearch/lib/opentelemetry/instrumentation/elasticsearch/patches/client.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'uri'
+
+module OpenTelemetry
+  module Instrumentation
+    module Elasticsearch
+      module Patches
+        # Module to prepend to Elasticsearch::Client for instrumentation
+        module Client
+          def perform_request(*args)
+            response = nil
+
+            tracer.in_span(
+              'elasticsearch.query',
+              attributes: tracing_attributes(*args),
+              kind: :client
+            ) do |span|
+              response = super(*args)
+              span.set_attribute('http.status_code', response.status)
+            end
+            response
+          end
+
+          private
+
+          def tracing_attributes(*args)
```

Comment:
I am not certain we want to add these attributes to the instrumentation since I do not think there aren't any [semantic conventions defined for Elasticsearch](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md).

It looks like the [Java Instrumention](https://github.com/search?q=org%3Aopen-telemetry+otel.instrumentation.elasticsearch.experimental-span-attributes&type=Code) uses a flag to enable experimental tags that are specific to that instrumentation. 

Reference: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2402 

I think that is something we should consider here as well and only limit the tags we capture by default. Thoughts?

cc: @fbogsany @mwear @robertlaurin @ericmustin @ahayworth

Reply:
tbh I'm not sure what the right approach is here regarding non semantic convention attributes when there aren't any specifics for elasticsearch in general. I will review a bit closer this wknd, I think Java's approach is reasonable but I also think it adds a higher config bar for the average user. maybe that's fine

one thing that is in the semantic conventions referenced by @arielvalentin is the attribute `db.system: elasticsearch` which I don't see in this PR. I think we should add that.

Another is that I think we should  make an attempt to quantize the `body`  and `url`, if possible, similar to what we've done for pg and mysql spans, as a config option.  We can leverage the datadog approach https://github.com/DataDog/dd-trace-rb/blob/339c1d2aea1752c43fb34d26a307938f24307ba5/lib/ddtrace/contrib/elasticsearch/patcher.rb#L95 or write our own, I don't have strong opinions here.

Lastly, i'm not sure what the right `span.name` value would be for elasticsearch here. Is a static value the correct approach?

Reply:
Thanks for the review @ericmustin 

|> attribute db.system: elasticsearch which I don't see in this PR. I think we should add that
sure, I will add it

Happy to take a look at the Java implementation and will wait for your review this weekend.

|> quantize the `body` and `url`
I was in doubt how to include it: "taking" the dd-trace [approach](https://github.com/DataDog/dd-trace-rb/blob/master/lib/ddtrace/contrib/elasticsearch/quantize.rb) or implement ours. 

Thanks!

Reply:
From the semantic conventions spec:

> In OpenTelemetry spans can be created freely and it’s up to the implementor to annotate them with attributes specific to the represented operation. 

The "implementor" here is us, I believe - and of course the text goes on to talk about why we should standardize as much as possible. But what this tells me is that we are not restricted to only adding the attributes already defined; but rather that we should adhere to the conventions whenever a convention exists.

Given that `elasticsearch` is mentioned in the database semantic conventions, I think that means we should ensure that we add any of the attributes defined there that are applicable (and of course, we should add any of the other attributes - such as the general attributes, rpc attributes, http attributes, etc - as applicable).

But I also think we should feel free to add any _other_ attributes that we think are necessary, even if they're not in the conventions already. We can use other language implementations as a guide, but they're not necessarily authoritative either. When we're ready to merge this, if it seems like there should be an update to the semantic conventions, we should do that - but I don't think that should block us from adding attributes in any way.

So - bottom line for me is "I think we should add them, but make sure they follow any / all applicable conventions that are already defined, and add definitions upstream to the spec if it makes sense."

FWIW - I don't think we should hide them behind a flag; that seems like a lot of friction to me. I don't think that's how the semantic conventions are supposed to be interpreted, personally.

Reply:
Thanks for sharing this perspective.

I feel like I want to have a more comprehensive discussion around how to work with experimental conventions that aren't well defined but I also don't want to hold up this PR to resolve it or come to a conclusion. I'll open a discussion and reference this PR so that we could build consensus around this and then provide explicit guidance in a contributors guide. 

That being said, I think something we all agree on is ensuring we follow any existing conventions so at a minimum we should ensure this PR included those; however for those attributes we find useful or not explicitly defined in the spec we should choose naming that resembles the format for existing examples which are [structured](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#examples) using this format  `db.<system>.<attribute>`  

Thoughts?

Reply:
> Thoughts?

👍